commons-numbers-examples/examples-jmh/src/main/java/org/apache/commons/numbers/examples/jmh/arrays/EuclideanNormAlgorithmPerformance.java

```diff
@@ -0,0 +1,235 @@
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+/**
+ * Execute benchmarks for the algorithms in the {@link EuclideanNormAlgorithms} class.
+ */
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Warmup(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@State(Scope.Benchmark)
+@Fork(value = 1, jvmArgs = {"-server", "-Xms512M", "-Xmx512M"})
+public class EuclideanNormAlgorithmPerformance {
+
+    /** Default number of samples used in each benchmark. */
+    private static final int DEFAULT_SAMPLES = 100_000;
+
+    /** Default length of vectors used in the benchmarks. */
+    private static final int DEFAULT_VECTOR_LEN = 100;
+
+    /** String indicating double exponents with very low negative values, likely to underflow. */
+    private static final String LOW = "low";
+
+    /** String indicating double exponents with mid-level values which will not overflow or underflow. */
+    private static final String MID = "mid";
+
+    /** String indicating double exponents with very high positive values, likely to overflow. */
+    private static final String HIGH = "high";
+
+    /** String indicating double exponents over a very wide range of values. */
+    private static final String FULL = "full";
+
+
+    /** Class providing input vectors for benchmarks.
+     */
+    @State(Scope.Benchmark)
+    public static class VectorArrayInput {
+
+        /** The number of samples. */
+        @Param(DEFAULT_SAMPLES + "")
+        private int samples;
+
+        /** The length of each vector. */
+        @Param(DEFAULT_VECTOR_LEN + "")
+        private int vectorLength;
+
+        /** The type of double values placed in the vector arrays. */
+        @Param({LOW, MID, HIGH, FULL})
+        private String type;
+
+        /** Array of input vectors. */
+        private double[][] vectors;
+
+        /** Get the input vectors.
+         * @return input vectors
+         */
+        public double[][] getVectors() {
+            return vectors;
+        }
+
+        /** Create the input vectors for the instance.
+         */
+        @Setup
+        public void createVectors() {
+            final UniformRandomProvider rng = RandomSource.create(RandomSource.XO_RO_SHI_RO_1024_PP);
+
+            int minExp;
+            int maxExp;
+
+            switch (type) {
+            case LOW:
+                minExp = -530;
+                maxExp = -510;
+                break;
+            case MID:
+                minExp = -10;
+                maxExp = +10;
+                break;
+            case HIGH:
+                minExp = +510;
+                maxExp = +530;
+                break;
+            default:
+                minExp = -550;
+                maxExp = +550;
+                break;
```

Comment:
I prefer here to throw in the default case statement. So if you pass parameters to JMH on the command line it will fail:
```
mvn package -Pexamples-jmh
java -jar target/examples-jmh.jar EuclideanNormAlgorithmPerformance -p vectorLength=10 -p type=foobar
```

This should error but instead runs with `type=FULL`.